### ISSUE TITLE
A question about data labels

### ISSUE THREAD

**Issue by luna9722** [umr-ds/feature_pyramid_fusion#1]
I read your research paper "Detection and segmentation of morphologically complex eukaryotic cells in fluorescence microscopy images via feature pyramid fusion" and I am very grateful that you published the dataset you used. I used this dataset for semantic cell segmentation with unet and I am planning to try mask rcnn for instance segmentation. In the paper you mentioned that the dataset provided include segmentation masks and bounding boxes but when i downloaded your dataset I couldn't get any information about the bounding boxes even in the metadata. Can you explain to me where can I find bounding boxes coordinates and how you fed your data to mask rcnn. I would be grateful if you can help me with this matter. Thank you in advance.

**Comment by nik-ko** [comment]
Thank you for pointing this out. Indeed the dataset structure is not decribed at all.
The [zipped tar archive](https://box.uni-marburg.de/index.php/s/N934NJi7IsvOphf/download) has the following structure:

```bash
├── images
│   └── {image_id}_mCherry.tif                 # Image of cells channel
│   └── {image_id}_DAPI.tif                    # Image of nucleus channel
├── mask                                    
│   └── {image_id}_mCherry.tif                 # Binary segmentation for whole image of cells channel
│   └── {image_id}_DAPI.tif                    # Binary segmentation for whole image of nucleus channel
└── object_masks
    └── {image_id}                             # Contains instance segmentation masks for all instances in image 
          └── {x}_{y}_{w}_{h}_mCherry.tif      # Segmentation mask for cells channel for a single instance
          └── {x}_{y}_{w}_{h}_DAPI.tif         # Segmentation mask for nucleus channel for a single instance
```

I.e. you can obtain the bounding box information for an instance of an image by parsing the filename (e.g. `object_masks/{image_id}/{x}_{y}_{w}_{h}_mCherry.tif`) and extract x,y,width and height. Or by reading the mask image and use OpenCV methods.

**Comment by luna9722** [comment]
Thanks for your response. Can I ask you about the data you fed to the mask rcnn repo, I tried your code and it didn't work for me, what is the shape of the images and the masks you fed the repository (did you use the original images and masks with shape 1024x1280 and only convert them to png images or did you do something else) I am really in need of your response. The research paper and the readme lacks information about this matter... Thank you in advance.

**Comment by nik-ko** [comment]
The training scripts are located in `samples/cells_and_cores`. You need to convert the images accordingly, and for better results, apply some more data augmentation.

**Comment by luna9722** [comment]
Thank you, I already figure it out, thank you very much for the research paper and the repo, it helped me a lot. I will be grateful if you can give me some other tips for cell instance segmentation (working on cells only) also didn't you try transfer learning using coco weights or it was useless ?

**Comment by luna9722** [comment]
#closed
